### PR TITLE
build(webpack) Upgrade webpack to v4

### PR DESCRIPTION
**Pull request by EugeneHlushko**
- Update webpack to v4
- use appropriate loaders
- fix issue with importing `Sidecar`
- fixed npm test build command wich was broken since rebuild changed output to `dist/`

====
- tested dev
- tested build

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,19 +77,18 @@
...
     "cross-env": "5.2.0",
-    "css-loader": "^0.28.5",
+    "css-loader": "^0.28.10",
     "directory-tree": "2.2.0",
     "directory-tree-webpack-plugin": "0.3",
     "duplexer": "^0.1.1",
     "eslint": "4.19.1",
     "eslint-loader": "^2.0.0",
     "eslint-plugin-markdown": "1.0.0-rc.0",
-    "extract-text-webpack-plugin": "^3.0.0",
```
Comment (EugeneHlushko) [inline package.json]: Looks like we dont use this anymore :)

File: package.json [hunk 2]
```diff
@@ -77,19 +77,18 @@
...
     "file-loader": "^1.1.11",
-    "fontgen-loader": "git://github.com/EugeneHlushko/fontgen-loader.git#a26a73843900ca4b518853952b1fc3c816103512",
+    "fontgen-loader": "git://github.com/EugeneHlushko/fontgen-loader.git#0.1.3",
```
Comment (EugeneHlushko) [inline package.json]: Updated the branch with vulnerability fix, master repo still stalled

File: src/components/Gitter/Gitter.jsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ export default class Gitter extends React.Component {
     if (isClient) {
       import('gitter-sidecar').then(Sidecar => {
         if (!sidecar) {
-          sidecar = new Sidecar({
+          sidecar = new Sidecar.default({
```
Comment (EugeneHlushko) [inline src/components/Gitter/Gitter.jsx]: This is the way now

File: src/components/Vote/Vote.scss [hunk 4]
```diff
@@ -1,6 +0,0 @@
-@import 'vars';
-@import 'mixins';
-@import 'functions';
-
-.vote {
```
Comment (EugeneHlushko) [inline src/components/Vote/Vote.scss]: It was empty, removed.

File: src/styles/partials/_functions.scss [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // Custom functions
 
 @import 'vars';
-@import '~modularscale-sass';
+@import '../../../node_modules/modularscale-sass/stylesheets/modularscale';
```
Comment (EugeneHlushko) [inline src/styles/partials/_functions.scss]: Couldnt get it to work without absolute path. Had tried different ways and the one from official readme.

Comment (jeremenichelli) [reply src/styles/partials/_functions.scss]: Hmm thinking if we are transpiling modules notation and it's causing issue, though I think we would see other issues.

File: webpack.prod.js [hunk 6]
```diff
@@ -22,14 +21,8 @@ const paths = [
 
 // Prod only config
 const prod = {
+  mode: 'production',
```
Comment (jeremenichelli) [inline webpack.prod.js]: 👏 👏 👏

**Comment by EugeneHlushko** [comment]
Thanks for reviewing @jeremenichelli ! Great stuff

**Pull request merged**